### ISSUE TITLE
Autocomplete feature for CLI options

### ISSUE THREAD

**Issue by adpaco-aws** [model-checking/kani#2203]
Having an autocomplete feature for CLI options would definitely improve the user experience.

A [quick search](https://www.google.com/search?q=autocomplete+clap+rust) suggests that `clap` is able to support autocomplete, although I haven't tried this yet myself. I'm also not sure if this will work since we use proxy binaries, but it's worth looking into.

If we can make it work somehow, I suspect there will be options that are easier to complete (e.g., `--output-format` which has a pre-fixed set of options) than others. For example, for `--harness` one would need to first run `kani-driver` in order to find the proofs. This may not be desired, so we should carefully evaluate the process for each option.

**Comment by celinval** [comment]
Yes, that would be awesome! I think at least enabling the auto complete for the flags would be amazing. For values that are not enums, I think it will be more complicated.

**Comment by celinval** [comment]
I just bumped into this enum to help with custom values: https://docs.rs/clap/4.2.7/clap/enum.ValueHint.html
